test/python/visualization/test_dag_drawer.py

```diff
@@ -36,6 +37,32 @@ def test_dag_drawer_invalid_style(self):
         """Test dag draw with invalid style."""
         self.assertRaises(VisualizationError, dag_drawer, self.dag, style="multicolor")
 
+    def test_dag_drawer_checks_filename_correct_format(self):
+        """filename must contain name and extension"""
+        try:
+            dag_drawer(self.dag, filename="aaabc")
+            self.fail("Expected error not raised!")
+        except InvalidFileError as exception_instance:
+            self.assertEqual(
```

Comment:
Here, and in the next test, a clearer way of writing these tests of exceptions is
```python
with self.assertRaisesRegex(InvalidFileError, "Parameter 'filename' must be in format 'name.extension'"):
    dag_drawer(self.dag, filename="aaabc")
```